sqlmesh/core/macros.py

```diff
@@ -120,6 +121,7 @@ def __init__(
         self._jinja_env: t.Optional[Environment] = jinja_env
         self.macros = {normalize_macro_name(k): v.func for k, v in macro.get_registry().items()}
         self._schema = MappingSchema(schema, dialect=dialect, normalize=False) if schema else {}
+        self._query_meta = query_meta
```

Comment:
i don't get how this is propogated to query.meta

Reply:
same dict refence

Reply:
but on line 182 you do self.query.meta, how is that related to this _query_meta

Reply:
when you create a renderer, you pass in `self.query` which is stored in the renderer's `self.expression` attribute and so `self._expression.meta` is the exact same dict object as `self.query.meta`, where the former one is passed to the macro evaluator and stored in `self._query_meta`